sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -1097,17 +1097,41 @@ case class StreamingDeduplicateWithinWatermarkExec(
 
   protected val extraOptionOnStateStore: Map[String, String] = Map.empty
 
-  private val eventTimeCol: Attribute = WatermarkSupport.findEventTimeColumn(child.output,
-    allowMultipleEventTimeColumns = false).get
-  private val delayThresholdMs = eventTimeCol.metadata.getLong(EventTimeWatermark.delayKey)
-  private val eventTimeColOrdinal: Int = child.output.indexOf(eventTimeCol)
```

Comment:
shall we simply make them `lazy val`?

Reply:
BTW, we can also check `isCanonicalizedPlan` and do something differently for canonilaized plan.

Reply:
Ah OK, that's good to know. Probably either way would simply work. Will try that out. Thanks!